tests/training/trainer_test.py

```diff
@@ -122,6 +123,28 @@ def test_should_stop_early_with_decreasing_metric(self):
         assert new_trainer._should_stop_early([.02, .3, .2, .1, .4, .4])  # pylint: disable=protected-access
         assert not new_trainer._should_stop_early([.3, .3, .2, .1, .4, .5])  # pylint: disable=protected-access
 
+    def test_should_stop_early_with_early_stopping_disabled(self):
+        # Increasing metric
+        trainer = Trainer(self.model, self.optimizer, self.iterator, self.instances,
+                          validation_dataset=self.instances, num_epochs=100,
+                          patience=None, validation_metric="+test")
+        decreasing_history = [float(i) for i in reversed(range(20))]
+        assert not trainer._should_stop_early(decreasing_history)  # pylint: disable=protected-access
+
+        # Decreasing metric
+        trainer = Trainer(self.model, self.optimizer, self.iterator, self.instances,
+                          validation_dataset=self.instances, num_epochs=100,
+                          patience=None, validation_metric="-test")
+        increasing_history = [float(i) for i in range(20)]
+        assert not trainer._should_stop_early(increasing_history)  # pylint: disable=protected-access
+
+    def test_should_stop_early_with_invalid_patience(self):
+        for patience in [0, -1, -2, 1.5, 'None']:
+            with pytest.raises(ConfigurationError, message='No ConfigurationError for patience={}'.format(patience)):
```

Comment:
This message seems wrong - shouldn't you be checking for "invalid value"?

Reply:
The default text message printed when the test fails is: 
``"Failed: DID NOT RAISE <class 'allennlp.common.checks.ConfigurationError'>"``
With mine would print:
``Failed: No ConfigurationError for patience=0``
I could replace "No" with "Did not raise" if you prefer.

Reply:
Ah, I was thinking of the `match` parameter to `pytest.raises`, not the `message` parameter.  Thanks =).  I think I've been using this wrong in other places in the code, too...